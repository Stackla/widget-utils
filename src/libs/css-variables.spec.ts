import getCSSVariables, { getTileSizeByWidget, trimHashValuesFromObject } from "./css-variables"

// Mock sdk object globally
const sdk = {
  getStyleConfig: jest.fn(),
  getInlineTileConfig: jest.fn(),
  getExpandedTileConfig: jest.fn()
}

// @ts-expect-error global properties are not typed
global.sdk = sdk

describe("Widget Functions", () => {
  // Tests for getTileSizeByWidget
  describe("getTileSizeByWidget", () => {
    it("should return medium size when inline_tile_size is not defined", () => {
      sdk.getStyleConfig.mockReturnValue({})

      const result = getTileSizeByWidget()
      expect(result).toEqual({ "--tile-size": "265.5px", "--tile-size-unitless": "265.5" })
    })

    it("should return correct tile size based on inline_tile_size", () => {
      sdk.getStyleConfig.mockReturnValue({ inline_tile_size: "small" })
      expect(getTileSizeByWidget()).toEqual({ "--tile-size": "173px", "--tile-size-unitless": "173" })

      sdk.getStyleConfig.mockReturnValue({ inline_tile_size: "medium" })
      expect(getTileSizeByWidget()).toEqual({ "--tile-size": "265.5px", "--tile-size-unitless": "265.5" })

      sdk.getStyleConfig.mockReturnValue({ inline_tile_size: "large" })
      expect(getTileSizeByWidget()).toEqual({ "--tile-size": "400px", "--tile-size-unitless": "400" })
    })
  })
  // Tests for getCSSVariables
  describe("getCSSVariables", () => {
    it("should return the correct CSS variables", () => {
      sdk.getStyleConfig.mockReturnValue({
        widget_background: "ffffff",
        text_tile_background: "000000",
        text_tile_link_color: "ff0000",
        text_tile_user_name_font_color: "0000ff",
        text_tile_user_handle_font_color: "00ff00",
        shopspot_btn_background: "ff00ff",
        shopspot_btn_font_color: "00ffff",
        cta_btn_background: "000000",
        cta_btn_font_color: "ffffff",
        cta_btn_font_size: "10",
        margin: 10,
        text_tile_font_size: 16,
        text_caption_paragraph_font_size: 12,
        text_tile_user_name_font_size: 16,
        text_tile_user_handle_font_size: null,
        shopspot_icon: null,
        expanded_tile_border_radius: 5,
        tile_tag_background: "D6D4D5"
      })

      sdk.getInlineTileConfig.mockReturnValue({
        show_timestamp: true,
        show_caption: false
      })

      sdk.getExpandedTileConfig.mockReturnValue({
        show_tags: false
      })

      // Check if the generated CSS variables match the snapshot
      const cssVariables = getCSSVariables()
      expect(cssVariables).toMatchSnapshot()
    })

    it("should return correct sizes with custom tile size settings", () => {
      sdk.getStyleConfig.mockReturnValue({})
      const tileSizeSettings = {
        tileSizeSettings: {
          small: "100px",
          medium: "200px",
          large: "300px"
        }
      }

      const result = getCSSVariables(tileSizeSettings)
      expect(result).toContain("--tile-size: 200px;")
    })

    it("should replace hashes with nothing in the returned object", () => {
      const value = trimHashValuesFromObject({
        widget_background: "#ffffff",
        text_tile_background: "#000000",
        text_tile_font_color: "#333333",
        text_tile_link_color: "#ff0000",
        text_tile_user_name_font_color: "#0000ff",
        text_tile_user_handle_font_color: "#00ff00",
        shopspot_btn_background: "#ff00ff",
        shopspot_btn_font_color: "#00ffff",
        margin: "10",
        text_tile_font_size: "16",
        text_caption_paragraph_font_size: "12",
        text_tile_user_name_font_size: "16",
        tile_tag_background: "#D6D4D5",
        auto_refresh: "",
        click_through_url: "",
        enable_custom_tiles_per_page: false,
        load_more_type: "scroll",
        name: "",
        plugin_instance_id: "",
        polling_frequency: "",
        rows_per_page: "",
        shopspot_btn_font_size: "",
        shopspot_icon: "",
        style: "",
        text_tile_user_handle_font_size: "",
        tile_background: "",
        tiles_per_page: "",
        minimal_tiles: "",
        type: "",
        widget_height: "",
        widget_loading_image: "",
        unavailable_products_behaviour: "always_show",
        dynamic_filter: "",
        dynamic_filter_fallback: {
          category: false,
          brand: false,
          custom: 0
        },
        inline_tile_size: "",
        inline_tile_border_radius: "",
        inline_tile_margin: "",
        expanded_tile_border_radius: "",
        map_content_feed_filter_id: "31915",
        map_content_feed_filter_name: "Map Feed",
        displayMode: "map_only",
        isScrollZoomDisabled: false,
        providerApiKey: "",
        providerLayer: "OpenStreetMap.BlackAndWhite",
        connectedContentWidgetId: "",
        defaultBounds: "-33.8336347704208,151.26371383666992,-33.876401620744,151.18646621704102",
        pinColor: "f7584c",
        pinTextColor: "ffffff",
        cta_btn_background: "000000",
        cta_btn_font_color: "ffffff",
        cta_btn_font_size: "10"
      })

      expect(value).toEqual({
        widget_background: "ffffff",
        text_tile_background: "000000",
        text_tile_font_color: "333333",
        text_tile_link_color: "ff0000",
        text_tile_user_name_font_color: "0000ff",
        text_tile_user_handle_font_color: "00ff00",
        shopspot_btn_background: "ff00ff",
        shopspot_btn_font_color: "00ffff",
        margin: "10",
        text_tile_font_size: "16",
        text_caption_paragraph_font_size: "12",
        text_tile_user_name_font_size: "16",
        tile_tag_background: "D6D4D5",
        auto_refresh: "",
        click_through_url: "",
        enable_custom_tiles_per_page: false,
        load_more_type: "scroll",
        name: "",
        plugin_instance_id: "",
        polling_frequency: "",
        rows_per_page: "",
        shopspot_btn_font_size: "",
        shopspot_icon: "",
        style: "",
        text_tile_user_handle_font_size: "",
        tile_background: "",
        tiles_per_page: "",
        minimal_tiles: "",
        type: "",
        widget_height: "",
        widget_loading_image: "",
        unavailable_products_behaviour: "always_show",
        dynamic_filter: "",
        dynamic_filter_fallback: {
          category: false,
          brand: false,
          custom: 0
        },
        inline_tile_size: "",
        inline_tile_border_radius: "",
        inline_tile_margin: "",
        expanded_tile_border_radius: "",
        map_content_feed_filter_id: "31915",
        map_content_feed_filter_name: "Map Feed",
        displayMode: "map_only",
        isScrollZoomDisabled: false,
        providerApiKey: "",
        providerLayer: "OpenStreetMap.BlackAndWhite",
        connectedContentWidgetId: "",
        defaultBounds: "-33.8336347704208,151.26371383666992,-33.876401620744,151.18646621704102",
        pinColor: "f7584c",
        pinTextColor: "ffffff",
        cta_btn_background: "000000",
        cta_btn_font_color: "ffffff",
        cta_btn_font_size: "10"
      })
    })
  })
})
