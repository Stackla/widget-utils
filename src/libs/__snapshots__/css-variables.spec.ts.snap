// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widget Functions getCSSVariables should return the correct CSS variables 1`] = `
"--widget-background: #ffffff;
--inline-tile-background: #000000;
--text-tile-background: #000000;
--shopspot-btn-background: #ff00ff;
--cta-button-background-color: #000000;
--tile-tag-background: #D6D4D5;
--text-tile-link-color: #ff0000;
--text-tile-user-handle-font-color: #00ff00;
--shopspot-btn-font-color: #00ffff;
--margin: 10px;
--text-tile-font-size: 16px;
--text-caption-paragraph-font-size: 16px;
--text-tile-user-name-font-size: 16px;
--text-tile-user-name-font-color: #0000ff;
--text-tile-user-handle-font-size: 12px;
--text-tile-font-color: #undefined;
--show-caption-inline: none;
--show-caption-inline-webkit: none;
--shopspot-icon: #000;
--tags-gap: 4px;
--cta-button-font-color: #ffffff;
--cta-button-font-size: 10px;
--expanded-tile-border-radius: 5px;
--tile-size: 265.5px;
--tile-size-unitless: 265.5;
--tile-width: 158px;
--inline-tile-border-radius: undefinedpx;
--inline-tile-margin: 10px;
--tags-display-inline: none;
--tags-display-expanded: none;
--story-expanded-display-bottom-section: 10px;
--shopspots-display-inline: none;
--shopspots-display-expanded: none;
--timephrase-display-inline: flex;
--share-icon-display-inline: none;"
`;
